src/asm_files.cpp

```
@@ -174,29 +172,28 @@ void relocate_map(ebpf_inst& inst, const std::string& symbol_name,
-
-        if (it == map_descriptors_offsets.end()) {
-            throw std::runtime_error("Bad reloc value (" + std::to_string(index) + "). " +
-                                     "Make sure to compile with -O2.");
+        const auto it = map_descriptors_offsets.find(symbol_name);
+        if (it != map_descriptors_offsets.end()) {
+            reloc_value = it->second;
         }
-        inst.imm = info.map_descriptors.at(it->second).original_fd;
     }
+    if (reloc_value >= info.map_descriptors.size()) {
+        throw std::runtime_error("Bad reloc value (" + std::to_string(reloc_value) + "). " +
```

cppcheck

[error] Syntax Error

(internalAstError)